### ISSUE TITLE
Upgrade to OpenSSL 1.1.1n

### ISSUE THREAD

**Issue by martinknafve** [hmailserver/hmailserver#425]
Released 2022-03-15

**Comment by RvdHout** [comment]
03-May-2022 | OpenSSL 1.1.1o is now available, including bug and security fixes

https://www.openssl.org/news/secadv/20220503.txt

**Comment by martinknafve** [comment]
I think 1.1.1o won't have any impact, right? The only fix I can see is:

> The c_rehash script does not properly sanitise shell metacharacters to
prevent command injection.  This script is distributed by some operating
systems in a manner where it is automatically executed.  On such operating
systems, an attacker could execute arbitrary commands with the privileges
of the script.

But this script is not bundled with hMailServer.

(I don't mind upgrading of course (I often do even though the OpenSSL vulnerabilities does not seem applicable) but looks like it would have no effect)

**Comment by RvdHout** [comment]
21-Jun-2022 | OpenSSL 1.1.1p is now available, including bug and security fixes

https://www.openssl.org/news/cl111.txt

```
 Changes between 1.1.1o and 1.1.1p [21 Jun 2022]

  *) In addition to the c_rehash shell command injection identified in
     CVE-2022-1292, further bugs where the c_rehash script does not
     properly sanitise shell metacharacters to prevent command injection have been
     fixed.

     When the CVE-2022-1292 was fixed it was not discovered that there
     are other places in the script where the file names of certificates
     being hashed were possibly passed to a command executed through the shell.

     This script is distributed by some operating systems in a manner where
     it is automatically executed.  On such operating systems, an attacker
     could execute arbitrary commands with the privileges of the script.

     Use of the c_rehash script is considered obsolete and should be replaced
     by the OpenSSL rehash command line tool.
     (CVE-2022-2068)
     [Daniel Fiala, Tomáš Mráz]

  *) When OpenSSL TLS client is connecting without any supported elliptic
     curves and TLS-1.3 protocol is disabled the connection will no longer fail
     if a ciphersuite that does not use a key exchange based on elliptic
     curves can be negotiated.
     [Tomáš Mráz]
```

**Comment by martinknafve** [comment]
Fixed in 5.6.9 build 2602.
